### DISCUSSION TITLE
question: How to resolve public chanell

### DISCUSSION THREAD

**Discussion by injuryzy** [gotd/td#1095] [Q&A]
u := "https://t.me/toupiao111"
			
		invite, err3 := client.API().MessagesCheckChatInvite(ctx, u)
			
		channel, err3 := client.API().ChannelsJoinChannel(ctx, &tg.InputChannel{
			ChannelID:  0,
			AccessHash: 0,
		})
加入这个公共群组 ，我需要ChannelID 和 AccessHash ，目前官网 api上面 没看见通过邀请链接获取的 这个该怎么实现呢

**Comment by injuryzy** [comment]
u := "toupiao111"
		invite, err3 := client.API().MessagesImportChatInvite(ctx, u)
这样也是一直报hash 错误

**Comment by ernado** [comment]
Please see userbot example:

https://github.com/gotd/td/blob/2acd01f5c7994c53b060bc28fdcad901d664128b/examples/userbot/main.go#L140-L147

Make sure that you are using resolver cache.

Also, please dont resolve instantly after logging in (e.g. right after start of the program) and be careful, I've got my test account banned while testing cache because I was resolving immediately after start.

**Reply by injuryzy** [reply]
我想通过公共群链接 用api的方式自动加入公共群里面，但是我不知道怎样加入进去，我用的ChannelsJoinChannel 这个方法，但是需要获取groupid 和hash ，没有在官方的api上面看到这个接口。MessagesImportChatInvite 用这个方法放入hash 也不能加入公共群组

比如这个公共群组 ：https://t.me/toupiao111

**Reply by lsy88** [reply]
> 我想通过群公共链接用api的方式自动加入公共群里面，但是我不知道怎样加入进去，我用的ChannelsJoinChannel这个方法，但是需要获取groupid和hash，在官方的api上面没有看到这个接口。MessagesImportChatInvite用这个方法里的hash也不能加入公共群
> 
> 比喻这个公共团体： https: [//t.me/toupiao111](https://t.me/toupiao111)

请问你找到加群的方法了吗
